kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +313,77 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps) throws KuraException {
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            if (prop.getKey().contains("802-1x.keystore.pid")) {
+                String keystorePid = (String) prop.getValue();
+
+                String interfaceName = prop.getKey().split("\\.")[2];
+                findAndDecodeCertificatesForInterface(interfaceName, modifiedProps, getKeystore(keystorePid));
+            }
+        }
+    }
+
+    private void findAndDecodeCertificatesForInterface(String interfaceName, Map<String, Object> modifiedProps,
+            KeystoreService keystoreService) {
+
+        if (keystoreService == null) {
+            logger.error("Cannot find keystore service for interface {}", interfaceName);
+            return;
+        }
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            String clientCertString = String.format("%s.config.802-1x.client-cert-name", interfaceName);
+            String caCertString = String.format("%s.config.802-1x.802-1x.ca-cert-name", interfaceName);
+            String privateKeyString = String.format("%s.config.802-1x.private-key-name", interfaceName);
```

Comment:
Why instantiate these strings here if they only depend on the `interfaceName`? They get instantiated for every property... this is a waste of clock cycles.

```suggestion
        final String clientCertString = String.format("%s.config.802-1x.client-cert-name", interfaceName);
        final String caCertString = String.format("%s.config.802-1x.802-1x.ca-cert-name", interfaceName);
        final String privateKeyString = String.format("%s.config.802-1x.private-key-name", interfaceName);

        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
```